src/hyperion/external_interaction/callbacks/ispyb_callback_base.py

```diff
@@ -12,16 +12,30 @@
     ISPYB_TRANSMISSION_FLUX_READ_PLAN,
     SIM_ISPYB_CONFIG,
 )
-from hyperion.parameters.internal_parameters import InternalParameters
+from hyperion.parameters.plan_specific.gridscan_internal_params import (
+    GridscanInternalParameters,
+)
+from hyperion.parameters.plan_specific.rotation_scan_internal_params import (
+    RotationInternalParameters,
+)
+
+if TYPE_CHECKING:
+    from hyperion.external_interaction.ispyb.store_in_ispyb import (
+        Store2DGridscanInIspyb,
+        Store3DGridscanInIspyb,
+        StoreRotationInIspyb,
+    )
 
 
 class BaseISPyBCallback(CallbackBase):
-    def __init__(self, parameters: InternalParameters):
-        """Subclasses should run super().__init__() with parameters, then set
+    def __init__(self) -> None:
+        """Subclasses should run super().__init__(), then in their start() method set
         self.ispyb to the type of ispyb relevant to the experiment and define the type
         for self.ispyb_ids."""
-        self.ispyb: StoreInIspyb
-        self.params = parameters
+        self.params: GridscanInternalParameters | RotationInternalParameters | None = (
```

Comment:
as above

Reply:
Again, which attributes have issues?

Reply:
sorry, I edited in which attributes have issues after you commented I guess. This one can't be fixed that easily, because of the ugly internal params parsing logic. it should get cleaned up in the external params refactor hopefully

Reply:
Yh, I still think there is a nicer way to do it. Let's revisit after that refactor.